UI/FleetButton.cpp

```diff
@@ -291,6 +313,40 @@ void FleetButton::LayoutIcons() {
         GG::Pt graphic_ul = middle - subtexture_sz / 2;
         m_selection_indicator->SizeMove(graphic_ul, graphic_ul + subtexture_sz);
     }
+
+    // refresh fleet button tooltip
+    if (m_fleet_blockaded) {
+        std::shared_ptr<Fleet> fleet;
+        std::shared_ptr<System> current_system;
+        std::map<int, bool> adjacent_system_IDs;
+        std::string available_exits = "";
+        int available_exits_count = 0;
+
+        if (!m_fleets.empty())
+            // can just pick first fleet because all fleets in system should have same exits
+            fleet = GetFleet(*m_fleets.begin());
```

Comment:
All fleets grouped under a fleet button should have the same exits available. If an armed fleet arrives from a different starlane, it will be opened as an exit for all fleets.